Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -3248,13 +3254,13 @@ HTML::SourceSnapshotParams Document::snapshot_source_snapshot_params() const
     // source policy container
     //     sourceDocument's policy container
 
-    return HTML::SourceSnapshotParams {
-        .has_transient_activation = verify_cast<HTML::Window>(HTML::relevant_global_object(*this)).has_transient_activation(),
-        .sandboxing_flags = m_active_sandboxing_flag_set,
-        .allows_downloading = !has_flag(m_active_sandboxing_flag_set, HTML::SandboxingFlagSet::SandboxedDownloads),
-        .fetch_client = relevant_settings_object(),
-        .source_policy_container = m_policy_container
-    };
+    auto snapshot_params = realm().create<HTML::SourceSnapshotParams>();
```

Comment:
Just something I noticed - we use `realm.create<T>` here and `heap().allocate<T>` in other places. THere is no realm initialization needed for this thing, so I wonder if we should just make it a GC::Cell instead of a `JS::Cell`. That way I guess this would give a compile error. Though, this doesn't solve that problem the other way around 🤔